docs/nd4j/templates/quickstart.md

```diff
@@ -0,0 +1,381 @@
+
+System.out.println(x.sum());
+System.out.println(x.min());
+System.out.println(x.max());
+/*
+[[    0.8621,    0.9224,    0.8407], 
+ [    0.1504,    0.5489,    0.9584]]
+4.2830
+0.1504
+0.9584
+*/
+```
+
+Provide a dimension argument to apply the operation across the specified dimension:
+
+```java
+int [] shape = {3,4};
+INDArray x = Nd4j.arange(12).reshape(shape);
+System.out.println(x);
+/*
+[[         0,    1.0000,    2.0000,    3.0000], 
+ [    4.0000,    5.0000,    6.0000,    7.0000], 
+ [    8.0000,    9.0000,   10.0000,   11.0000]]
+*/        
+
+System.out.println(x.sum(0)); // Sum of each column.
+//[   12.0000,   15.0000,   18.0000,   21.0000]
+
+System.out.println(x.min(1)); // Min of each row
+//[         0,    4.0000,    8.0000]
+
+System.out.println(x.cumsum(1)); // cumulative sum across each row,
+/*
+[[         0,    1.0000,    3.0000,    6.0000], 
+ [    4.0000,    9.0000,   15.0000,   22.0000], 
+ [    8.0000,   17.0000,   27.0000,   38.0000]]
+*/
+
+```
+
+<!--- The numpy quickstart calls them Universal Functions. -->
+### Transform operation
+Nd4j provides familiar mathematical functions such as sin, cos, and exp. These are called transform operations. The result is returned as an INDArray.
+
+```java
+INDArray x = Nd4j.arange(3);
+System.out.println(x);
+// [         0,    1.0000,    2.0000]
+System.out.println(exp(x));
+// [    1.0000,    2.7183,    7.3891]
+System.out.println(sqrt(x));
+// [         0,    1.0000,    1.4142]
+```
+
+You can check out a complete list of transform operations in the [Javadoc](https://deeplearning4j.org/api/latest/org/nd4j/linalg/api/ops/impl/transforms/package-summary.html )
+
+
+### Matrix multiplication
+We have already seen the element wise multiplcation in the basic operations. The other Matrix operations have their own methods:
+
+```java
+int[] shape = {3, 4};
+INDArray x = Nd4j.arange(12).reshape(shape);
+int[] shape2 = {4, 3};
+INDArray y = Nd4j.arange(12).reshape(shape2);
+
+System.out.println(x);
+/*
+[[         0,    1.0000,    2.0000,    3.0000], 
+ [    4.0000,    5.0000,    6.0000,    7.0000], 
+ [    8.0000,    9.0000,   10.0000,   11.0000]]
+*/
+
+System.out.println(y);
+/*
+[[         0,    1.0000,    2.0000], 
+ [    3.0000,    4.0000,    5.0000], 
+ [    6.0000,    7.0000,    8.0000], 
+ [    9.0000,   10.0000,   11.0000]]
+*/
+
+System.out.println(x.mmul(y));  // matrix product.
+/*
+[[   42.0000,   48.0000,   54.0000], 
+ [  114.0000,  136.0000,  158.0000], 
+ [  186.0000,  224.0000,  262.0000]]
+*/
+
+// dot product.
+INDArray x = Nd4j.arange(12);
+INDArray y = Nd4j.arange(12);
+System.out.println(dot(x, y));  
+//506.0000
+```
+
+### Indexing, Slicing and Iterating
+Indexing, Slicing and Iterating is harder in Java than in Python. 
+To retreive individual values from an INDArray you can use the `getDouble`, `getFloat` or `getInt` methods. INDArrays cannot be indexed like Java arrays. You can get a Java array from an INDArray using `.data().asDouble()`. 
```

Comment:
`.data().asDouble()` etc isn't something I want in the quickstart, it's confusing for new users due to views and f/c order (not to mention permuted arrays).
Instead, direct users to toFloatVector(), toDoubleMatrix() etc